apollo-router/src/query_planner/caching_query_planner.rs

```diff
@@ -693,10 +698,28 @@ impl ValueType for Result<QueryPlannerContent, Arc<QueryPlannerError>> {
     }
 }
 
+// Add this helper function
+fn should_cache_error(error: &QueryPlannerError) -> bool {
+    match error {
+        // Temporary errors - don't cache
+        QueryPlannerError::JoinError(_) => false,
```

Comment:
In our huddle, it was mentioned that this may happen when the compute job queue is full. Is that true? And if so, it should be explained here or as a comment on the JoinError branch.